pkg/metrics/metrics.go

```diff
@@ -0,0 +1,143 @@
+package metrics
+
+import (
+	"github.com/golang/glog"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	machineinformers "github.com/openshift/cluster-api/pkg/client/informers_generated/externalversions/machine/v1beta1"
+	machinelisters "github.com/openshift/cluster-api/pkg/client/listers_generated/machine/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var (
+	// MachineCountDesc is a metric about machine object count in the cluster
+	MachineCountDesc = prometheus.NewDesc("mapi_machine_items_count", "Count of machine objects currently at the apiserver", nil, nil)
+	// MachineSetCountDesc Count of machineset object count at the apiserver
+	MachineSetCountDesc = prometheus.NewDesc("mapi_machineset_items_count", "Count of machinesets at the apiserver", nil, nil)
+	// MachineInfoDesc is a metric about machine object info in the cluster
+	MachineInfoDesc = prometheus.NewDesc("mapi_machine_created", "Timestamp of the mapi managed Machine creation time", []string{"name", "namespace", "spec_provider_id", "node", "api_version"}, nil)
+	// MachineSetInfoDesc is a metric about machine object info in the cluster
+	MachineSetInfoDesc = prometheus.NewDesc("mapi_machineset_created", "Timestamp of the mapi managed Machineset creation time", []string{"name", "namespace", "api_version"}, nil)
+
+	// MachineSetStatusAvailableReplicasDesc is the information of the Machineset's status for available replicas.
+	MachineSetStatusAvailableReplicasDesc = prometheus.NewDesc("mapi_machine_set_status_available_replicas", "Information of the mapi managed Machineset's status for available replicas", []string{"name", "namespace"}, nil)
```

Comment:
(nit) kube-state-metrics uses names like `<base name>_<status>` for similar metrics (eg `kube_deployment_status_replicas_available`, `kube_deployment_status_replicas_updated`, ...). It might be good to follow the same convention (IMO it is easier this way as the autocomplete would list the metrics together).